packages/components/CONTRIBUTING.md

```diff
@@ -509,7 +524,23 @@ Given a component folder (e.g. `packages/components/src/unit-control`):
 	4. Use existing HTML types when possible? (e.g. `required` for an input field?)
 	5. Use the `CSSProperties` type where it makes sense.
 	6. Extend existing components’ props if possible, especially when a component internally forwards its props to another component in the package.
-	7. Use `WordPressComponent` type if possible.
+	7. If the component forwards its `...restProps` to an underlying element, you should use the `WordPressComponentProps` type for the component's props:
+
+		```jsx
+		import type { WordPressComponentProps } from '../ui/context';
+		import type { ComponentOwnProps } from './types';
+
+		function UnconnectedMyComponent(
+			// The resulting type will include:
```

Comment:
Should we add the stuff about the `ref`?

Reply:
Pushed some changes in https://github.com/WordPress/gutenberg/pull/41669/commits/cb34385e89d3ff7885c2ef9fb30570a1f0cb8bb5, let me know what you think!

Main changes:

- moved down the point re. making sure that there's a name export
- added a new point re. ref forwarding (either via `forwardRef` or via `contextConnect`)
- updated index of remaining points as needed